source/grid/grid_tools.cc

```diff
@@ -4915,6 +4915,208 @@ namespace GridTools
 
 
 
+  template <int dim, int spacedim>
+  std::tuple< std::vector<typename Triangulation<dim, spacedim>::active_cell_iterator >, std::vector< std::vector< Point<dim> > >, std::vector<std::vector<unsigned int> > >
+  compute_point_locations(const Cache<dim,spacedim>                                         &cache,
+                          const std::vector<Point<spacedim> >                               &points,
+                          const double                                                      &distance_factor)
+  {
+    Assert(distance_factor >= 0.5,
+           ExcMessage("distance_factor value must be >= 0.5"));
+
+    std::tuple< std::vector<typename Triangulation<dim, spacedim>::active_cell_iterator >, std::vector< std::vector< Point<dim> > >, std::vector<std::vector<unsigned int> > >
+    cell_qpoint_map;
+
+    // How many points are here?
+    const unsigned int np = points.size();
+
+    typename Triangulation<dim, spacedim>::active_cell_iterator cell = cache.get_triangulation().begin_active();
+    // Now the easy case.
+    if (np==0) return cell_qpoint_map;
+
+    // Keep track of the points we
+    // found
+    std::vector<bool> point_flags(np, false);
+
+    // Set this to true until all
+    // points have been classified
+    bool left_over = true;
+
+    // Current cell center and diameter
+    Point<spacedim> cell_center = cell->center();
+    double cell_diameter = cell->diameter()*(distance_factor + std::numeric_limits<double>::epsilon() );
+
+    // see if the point is
```

Comment:
please keep in-code comment formatted to larger widths (see other files, i think it's around 80 chars)

Reply:
same below.